docs/architecture/adr-033-protobuf-inter-module-comm.md

```diff
@@ -0,0 +1,371 @@
+that module dependencies are resolved at startup, the `Configurator.RequireServer` method should be added. The `ModuleManager`
+will make sure that all dependencies declared with `RequireServer` can be resolved before the app starts. An example
+module `foo` could declare it's dependency on `x/bank` like this:
+
+```go
+package foo
+
+func (am AppModule) RegisterServices(cfg Configurator) {
+  cfg.RequireServer((*bank.QueryServer)(nil))
+  cfg.RequireServer((*bank.MsgServer)(nil))
+}
+```
+
+### Security Considerations
+
+In addition to checking for `ModuleKey` permissions, a few additional security precautions will need to be taken by
+the underlying router infrastructure.
+
+#### Recursion and Re-entry
+
+Recursive or re-entrant method invocations pose a potential security threat. This can be a problem if Module A
+calls Module B and Module B calls module A again in the same call.
+
+One basic way for the router system to deal with this is to maintain a call stack which prevents a module from
+being referenced more than once in the call stack so that there is no re-entry. A `map[string]interface{}` table
+in the router could be used to perform this security check.
+
+#### Queries
+
+Queries in Cosmos SDK are generally un-permissioned so allowing one module to query another module should not pose
+any major security threats assuming basic precautions are taken. The basic precaution that the router system will
+need to take is making sure that the `sdk.Context` passed to query methods does not allow writing to the store. This
+can be done for now with a `CacheMultiStore` as is currently done for `BaseApp` queries.
+
+### Internal Methods
+
+In many cases, we may wish for modules to call methods on other modules which are not exposed to clients at all. For this
+purpose, we add the `InternalServer` method to `Configurator`:
+
+```go
+type Configurator interface {
+   MsgServer() grpc.Server
+   QueryServer() grpc.Server
+   InternalServer() grpc.Server
+}
+```
+
+As an example, x/slashing's Slash must call x/staking's Slash, but we don't want to expose x/staking's Slash to end users
+and clients.
+
+This wound also require creating a corresponding `internal.proto` file with a protobuf service in the given module's
+proto package.
+
+Services registered against `InternalServer` will be callable from other modules but not by external clients.
+
+An alternative solution to internal-only methods could involve hooks / plugins as discussed [here](https://github.com/cosmos/cosmos-sdk/pull/7459#issuecomment-733807753).
+A more detailed evaluation of a hooks / plugin system will be addressed later in follow-ups to this ADR or as a separate
+ADR.
+
+### Authorization
+
+By default, the inter-module router requires that messages are sent by the first signer returned by `GetSigners`. The
+inter-module router should also accept authorization middleware such as that provided by [ADR 030](https://github.com/cosmos/cosmos-sdk/pull/7105).
```

Comment:
Now that the PR is merged, can you reference the adr in master: https://github.com/cosmos/cosmos-sdk/blob/master/docs/architecture/adr-030-authz-module.md ?

BTW, that adr saws it was accepted 2021-11-13 (in 10 months), I think you need to swap a 1 for a 0.